addons/microsoft_calendar/models/calendar.py

```diff
@@ -74,13 +74,31 @@ def create(self, vals_list):
         if self._check_microsoft_sync_status() and not notify_context and recurrency_in_batch:
             self._forbid_recurrence_creation()
 
-        # for a recurrent event, we do not create events separately but we directly
-        # create the recurrency from the corresponding calendar.recurrence.
-        # That's why, events from a recurrency have their `need_sync_m` attribute set to False.
-        return super(Meeting, self.with_context(dont_notify=notify_context)).create([
-            dict(vals, need_sync_m=False) if vals.get('recurrence_id') or vals.get('recurrency') else vals
-            for vals in vals_list
-        ])
+        # Create events separately since the sender user may differ between events.
+        events = self.env['calendar.event']
+        for vals in vals_list:
+            # Get sender user id from vals_list and then load the sender user from database.
+            # If sender user differs from current, check if its sync is active for creating events as organizer.
+            # Additionally, get the partner_ids from values and notify user if the pretended organizer is not on the list.
+            sender_user_id = vals['user_id'] if 'user_id' in vals else self.env.user.id
+            sender_user = self.env['res.users'].search([('id', '=', sender_user_id)])
```

Comment:
Searching inside a loop is bad performance-wise as each time you're doing a cache-read/fetching from the DB. Consider doing this before the loop:

```py
user_ids = [vals['user_id'] for vals in vals_list if 'user_id' in vals]
users = self.env['res.users'].browse(user_ids)
users_by_id = {user.id: user for user in users}
```

and inside the loop you can get the user by doing `sender_user = users_by_id[vals['user_id']] if 'user_id' in vals else self.env.user`